pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -639,17 +639,30 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 	// 1) New clusters where we are starting one deployment at a time. We only need to check for quorum once when we add a new mon.
 	// 2) Clusters being restored where no mon deployments are running. We need to start all the deployments before checking quorum.
 	onlyCheckQuorumOnce := false
-	deployments, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
+	deployments, derr := c.context.Clientset.AppsV1().Deployments(c.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)})
+	pods, perr := c.context.Clientset.CoreV1().Pods(c.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)})
+
+	if derr != nil || perr != nil {
+		if kerrors.IsNotFound(derr) {
 			logger.Infof("0 of %d expected mon deployments exist. creating new deployment(s).", len(mons))
 			onlyCheckQuorumOnce = true
 		} else {
-			logger.Warningf("failed to list mon deployments. attempting to continue. %v", err)
+			logger.Warningf("failed to list mon deployments. attempting to continue. %v", derr)
+			return derr
+		}
+		if kerrors.IsNotFound(perr) {
+			logger.Infof("0 of %d expected mon pod exist. creating or updating deployment(s).", len(mons))
+			onlyCheckQuorumOnce = true
+		} else {
+			logger.Warningf("failed to list mon pods. attempting to continue. %v", perr)
+			return perr
 		}
 	} else if len(deployments.Items) < len(mons) {
 		logger.Infof("%d of %d expected mon deployments exist. creating new deployment(s).", len(deployments.Items), len(mons))
 		onlyCheckQuorumOnce = true
+	} else if len(deployments.Items) != len(pods.Items) {
```

Comment:
I don't think we can compare the number of deployments to number of pods. There may be situations where the counts won't match because the pod is being restarted and the new pod is still pending while the old one is terminating.  Let's just add the new check for the case where no mon pods exist. That seems sufficient.

Reply:
cool added the check for this.